ReSharper.FSharp/src/FSharp/FSharp.Psi.Services/src/Util/FSharpBindUtil.fs

```diff
@@ -13,13 +13,17 @@ open JetBrains.ReSharper.Psi.ExtensionsAPI.Tree
 open JetBrains.ReSharper.Psi.Tree
 open JetBrains.ReSharper.Psi.Util
 
+let addOpensIfNeeded (reference: FSharpSymbolReference) (declaredElement: IClrDeclaredElement) opName =
+    let resolveExpr = not (reference.GetElement() :? ITypeReferenceName)
+    if not (FSharpResolveUtil.resolvesToQualified declaredElement reference resolveExpr opName) then
+        addOpens reference declaredElement |> ignore
+
 // todo: we modify the tree and then ask FCS. What tree is used? Do we wait for the new analysis?
 let bindDeclaredElementToReference (context: ITreeNode) (reference: FSharpSymbolReference)
         (declaredElement: IClrDeclaredElement) opName =
-    reference.SetRequiredQualifiers(declaredElement, context)
 
-    if not (FSharpResolveUtil.resolvesToQualified declaredElement reference true opName) then
-        addOpens reference declaredElement |> ignore
+    reference.SetRequiredQualifiers(declaredElement, context)
+    addOpensIfNeeded reference declaredElement opName
```

Comment:
Could you explain this change, please?

Reply:
I tried to divide the operations into those that trigger resolve and those that don't, so I needed to call the addition of opens separately.  I moved that into a separate function and called it where necessary.